provider/common/bootstrap_test.go

```diff
@@ -135,6 +135,55 @@ func (s *BootstrapSuite) TestCannotStartInstance(c *gc.C) {
 	c.Assert(err, gc.ErrorMatches, "cannot start bootstrap instance: meh, not started")
 }
 
+func (s *BootstrapSuite) TestBootstrapSeries(c *gc.C) {
+	s.PatchValue(&jujuversion.Current, coretesting.FakeVersionNumber)
+	s.PatchValue(&series.HostSeries, func() string { return "precise" })
+	stor := newStorage(s, c)
+	checkInstanceId := "i-success"
+	checkHardware := instance.MustParseHardware("arch=ppc64el mem=2T")
+
+	startInstance := func(
+		_ string, _ constraints.Value, _ []string, _ tools.List, icfg *instancecfg.InstanceConfig,
+	) (
+		instance.Instance, *instance.HardwareCharacteristics, []network.InterfaceInfo, error,
+	) {
+		return &mockInstance{id: checkInstanceId}, &checkHardware, nil, nil
+	}
+	var mocksConfig = minimalConfig(c)
+	var getConfigCalled int
+	getConfig := func() *config.Config {
+		getConfigCalled++
+		return mocksConfig
+	}
+	setConfig := func(c *config.Config) error {
+		mocksConfig = c
+		return nil
+	}
+
+	env := &mockEnviron{
+		storage:       stor,
+		startInstance: startInstance,
+		config:        getConfig,
+		setConfig:     setConfig,
+	}
+	ctx := envtesting.BootstrapContext(c)
+	bootstrapSeries := "utopic"
+	result, err := common.Bootstrap(ctx, env, environs.BootstrapParams{
+		BootstrapSeries: bootstrapSeries,
+		AvailableTools: tools.List{
+			&tools.Tools{
+				Version: version.Binary{
+					Number: jujuversion.Current,
+					Arch:   arch.HostArch(),
+					Series: bootstrapSeries,
+				},
+			},
+		}})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(result.Arch, gc.Equals, "ppc64el") // based on hardware characteristics
+	c.Assert(result.Series, gc.Equals, bootstrapSeries)
```

Comment:
Change the prior 2 asserts to `.Check` so that we get all errors in a failing test. `.Assert` needs to be kept on the err as a gate.